lang/py/avro/schema.py

```diff
@@ -601,6 +646,12 @@ def to_json(self, names=None):
             names.names[self.fullname] = self
             return names.prune_namespace(self.props)
 
+    def to_canonical_json(self, names):
```

Comment:
Should names be optional?

Reply:
Yes. Addressed in eacfdd6a0cd6de9b4e15701d08d9e994f09be310